pkg/cmd/roachtest/metrics/api.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+// Package metrics provides a fluent API for querying and asserting on metrics
+// in roachtests.
+package metrics
+
+import (
+	"time"
+)
+
+// MetricQuery represents a query that can be executed or asserted on
+type MetricQuery interface {
+	// Query building methods
+	ForService(serviceName string) MetricQuery
+	ForNode(nodeID string) MetricQuery
+	WithLabel(name, value string) MetricQuery
+	OverLast(duration string) MetricQuery
+	Between(start, end time.Time) MetricQuery
+	Sum() MetricQuery
+	Avg() MetricQuery
+	Max() MetricQuery
+	Rate() MetricQuery
+
+	// Execution methods
+	Scan(dest interface{}) error
+	Value() (float64, error)
+
+	// Assertion methods
+	AssertHasValue(expected float64) error
+	AssertHasValueAtLeast(threshold float64) error
+	AssertHasValueAtMost(threshold float64) error
+	AssertHasRateAtLeast(threshold float64) error
+	AssertHasRateAtMost(threshold float64) error
+	AssertHasPercentile(percentile int, threshold float64) error
+}
+
+// MetricsAPI provides methods to query and assert on metrics
+type MetricsAPI interface {
```

Comment:
I think we should also think about consolidating `clusterStatCollector.CollectInterval` and `clusterStatCollector.CollectPoint`. As a complete example of that API, look at `cdcTester.startStatsCollection` (in `cdc.go`).

There is also a refactored example of it in [1], namely `QueryRange`. If a user query doesn't map into any of the above, we'll need to provide an "escape hatch"; both `QueryRange` and `Query` could serve that purpose.

[1] https://github.com/cockroachdb/cockroach/pull/138252/files#diff-0ac943ad345ac75d52f046d8771553b6a66601697157c62ac547346e8039e089R141